File: OnlineBookstore (Oracle ORDBMS)/cgi-bin/BookStore2.java
// Import the following packages to use JDBC.
import  java.sql.*;
import  java.sql.DatabaseMetaData;
import  java.io.*;
import  oracle.sql.*;
import  oracle.jdbc.*;
import  oracle.jdbc.pool.OracleDataSource;
import java.lang.Object;
import java.lang.StringBuilder;


class  BookStore2 {
    public static void  main( String args[ ] ) throws SQLException {
        String user     = "C##mdsaifur.rahman.1";
        String password = "rahman9451";
        String database = "65.52.222.73:1521/cdb1";

        // Open an OracleDataSource and get a connection.
        OracleDataSource ods = new OracleDataSource( );
        ods.setURL     ( "jdbc:oracle:thin:@" + database );
        ods.setUser    ( user );
        ods.setPassword( password );
        Connection conn = ods.getConnection( );


        String     cmd, cmdS, cmdBS, cmdBS1, cmdNT;
        Statement  stmt1 = conn.createStatement( );

        Statement  stmt = conn.createStatement( );
        ResultSet  rset,rset2;
        //ResultSet rset1;

        if ( args[0].equals( "insert" ) ) {

            for ( int i=4;  i < args.length;  i++ ){

                cmdS = "insert /*+ ignore_row_on_dupkey_index(subject_table2(subject_name)) */ into subject_table2 ";
                cmdS += "values(subject_id.NEXTVAL, '" + args[i].trim( ) +"')";
                System.out.println( cmdS );
                stmt.execute( cmdS );
            }

            cmd  = "insert /*+ ignore_row_on_dupkey_index(book_table(title)) */ into book_table ";
            cmd += "values ('" + args[1].trim( );
            cmd += "','"+args[2].trim( );
            cmd += "', " + args[3].trim( );
            cmd += " , subject_type(";
            //System.out.println( cmd );
            //stmt.execute( cmd );


            //cmdBS = "select s.subject_id,regexp_replace( s.subject_id,',[^[:digit:]]*$') from subject_table2 s where (" ;
            cmdBS = "select s.subject_id from subject_table2 s where (" ;
             if ( args.length == 0 ) { cmd += "subject_name=' ')";}
             else {
             for ( int i=4; i < args.length; i++){

                        if (i == 4 ){ cmdBS += " s.subject_name ='"+ args[i].trim( );}

                        else { cmdBS += "' or  s.subject_name ='"+ args[i].trim( );}}//}

            cmdBS += "')";
             System.out.println( cmdBS );
             rset = stmt1.executeQuery(cmdBS);
             int s = 0;
             // StringBuilder strItems = new StringBuilder();
            while(rset.next()){//{
                if (s==0){
                    cmd += "subject_t2(" + rset.getString(1)+")";
                  s+=1;
                }
                else
                  {
                      cmd += ",subject_t2(" + rset.getString(1)+")";
                }

            }

             rset.close();
            cmd += ")) ";
            System.out.println( cmd );
            stmt.execute( cmd );


             }}
        else if ( args[0].equals( "List All Data" ) ) {
            cmd  = "select distinct b.title from book_table b ";
            System.out.println( cmd+"<br/><br/>" );
            rset = stmt.executeQuery( cmd );
            while ( rset.next( ) ){

                System.out.print(" <input type='radio'/><font color='0B6A34'>"+"Books Title: </font>"+ "<a href='http://undcemcs02.und.edu/~mdsaifur.rahman.1/cgi-bin/hlink2.cgi?title=" );
                System.out.print( rset.getString(1)+"'>"+rset.getString(1) +"</a><br/><br/>" );
            }
            rset.close( );

            cmd  = "select distinct s.subject_name from subject_table2 s ";
            System.out.println( cmd+"<br/><br/>" );
            rset = stmt.executeQuery( cmd );
            while ( rset.next( ) ){

                System.out.print(" <input type='radio'/><font color='0B6A34'>"+"Subject Name: </font>"+ "<a href='http://undcemcs02.und.edu/~mdsaifur.rahman.1/cgi-bin/hlinkSub2.cgi?subject_name=" );
                System.out.print( rset.getString(1)+"'>"+rset.getString(1)+"</a> <br/><br/>" );
            }
            rset.close( );

            cmd  = "select distinct c.cust_name from cust_table c ";
            System.out.println( cmd+"<br/><br/>" );
            rset = stmt.executeQuery( cmd );
            while ( rset.next( ) ){

                System.out.print(" <input type='radio'/><font color='0B6A34'>"+"Customer Name: </font>"+ "<a href='http://undcemcs02.und.edu/~mdsaifur.rahman.1/cgi-bin/hlinkCust.cgi?cust_name=" );
                System.out.print( rset.getString(1)+"'>"+rset.getString(1)+"</a> <br/><br/>" );
            }
            rset.close( );
}

        else if ( args[0].equals( "List All Books" ) ) {
            cmd  = "select distinct b.title from book_table b ";
            System.out.println( cmd+"<br/><br/>" );
            rset = stmt.executeQuery( cmd );
            while ( rset.next( ) ){

                System.out.print(" <input type='radio' name='title' value='"+rset.getString(1)+"'/><font color='0B6A34'>"+"Books Title:"+ "<a href='http://undcemcs02.und.edu/~mdsaifur.rahman.1/cgi-bin/hlink2.cgi?title=" );
                System.out.print( rset.getString(1)+"'>"+rset.getString(1) +"</a> </font><br/><br/>" );
                //System.out.print( "<br/> <font color='#3366CC'>"+"Price: $</font>"+rset.getString(2)+ "</form><br /><br/>" );
            }
            rset.close( );}
        else if ( args[0].equals( "List All Subjects" ) ) {
            cmd  = "select distinct s.subject_name from subject_table2 s";
            System.out.println( cmd+"<br/><br/>" );
            rset = stmt.executeQuery( cmd );
            while ( rset.next( ) ){
                System.out.print("<input type='checkbox'/><font color='0B6A34'>"+"Subject Name: "+"</font>" + "<a href='http://undcemcs02.und.edu/~mdsaifur.rahman.1/cgi-bin/hlinkSub2.cgi?subject_name=" );
                   System.out.print( rset.getString(1)+" '>" + rset.getString(1)+"</a>" +"<br /><br/>" );
                 }
            // Close the ResultSet and Statement.
            System.out.print( "<input type='submit' name='act' value='search' /> </form><br /><br/>");
            rset.close( );
           }

            else if ( args[0].equals("Master Clear"))  {

                 cmd  = "delete from book_table bt";
                 System.out.println( cmd );
                 stmt.execute( cmd );


                cmdS = "delete from subject_table2 st";
                System.out.println( cmdS );
                stmt.execute( cmdS );


                cmdBS = "delete from cust_table ct";
                System.out.println( cmdBS );
                stmt.execute( cmdBS );
            }

        else if ( args[0].equals( "Display Customer Data" ) ) {
            cmd  = "select distinct c.cust_name from cust_table c";
            System.out.println( cmd+"<br/><br/>" );
            rset = stmt.executeQuery( cmd );
            while ( rset.next( ) ){
                System.out.print("<input type='checkbox' /><font color='0B6A34'>"+"Customer Name: "+"</font>" + "<a href='http://undcemcs02.und.edu/~mdsaifur.rahman.1/cgi-bin/hlinkCust.cgi?cust_name=" );
                System.out.print( rset.getString(1)+"'>" + rset.getString(1)+"</a>" +"<br /><br/>" );
            }
            // Close the ResultSet and Statement.
            System.out.print( "<input type='submit' name='act' value='search' /> </form><br /><br/>");
            rset.close( );
        }

        stmt.close( );

        // Close the Connection.
        conn.close( );
    }
}

File: OnlineBookstore (Oracle ORDBMS)/cgi-bin/BookStore2.pl
#!/usr/bin/perl
use CGI;
$query   = new CGI;
$act     = $query->param('act');
$radio = $query->param( "radio" );
$ISBN = $query->param( "ISBN" );
$title = $query->param( "title" );
$price = $query->param( "price" );
$subject_name = $query->param( "subject_name" );


if ( $act eq "Submit" ) {
    print ( "Content-type: text/html\n\n");
# Use "here-doc" syntax.
    print <<EndofHTML;
  <html>
   <body background = "http://undcemcs02.und.edu/~mdsaifur.rahman.1/bg/bground.png">
    <table width="100%" height="80%">
     <tr>
      <td align="center">
       <font size="+0"><b>
EndofHTML

  # Compose a Java command.
$subject_name =~ s/^\s*(\S*)\s*$/$1/;
$subject_name =~ s/;|>|>>|<|\*|\?|\&|\|//g;

    if ( $radio eq "insert" ) {
        $cmd = "/usr/bin/java -Djava.security.egd=file:/dev/./urandom BookStore2 'insert' '";

        $cmd .= $ISBN . "' '" . $title."' '".$price . "' " ;
        my @subject_name = $query->param( 'subject_name' );
        foreach my $subject_name (@subject_name) { $cmd .= "' ".$subject_name. "' "; }
        print ( $cmd . "<br />\n\n" );
        system( $cmd );}
    elsif ( $radio eq "List All Data" ) {
        $cmd = "/usr/bin/java -Djava.security.egd=file:/dev/./urandom BookStore2 'List All Data'";
    # print ( $cmd . "\n\n" );
        system( $cmd );
    }

    elsif ( $radio eq "List All Books" ) {
        $cmd = "/usr/bin/java -Djava.security.egd=file:/dev/./urandom BookStore2 'List All Books'";
    # print ( $cmd . "\n\n" );
        system( $cmd );
    }
    elsif ( $radio eq "List All Subjects" ) {

        $cmd = "/usr/bin/java -Djava.security.egd=file:/dev/./urandom BookStore2 'List All Subjects'";

        system( $cmd );
    }

    elsif ( $radio eq "Master Clear" ) {
        $cmd  = "/usr/bin/java -Djava.security.egd=file:/dev/./urandom BookStore2 'Master Clear'";
        print ( $cmd . "\n\n" );
        system( $cmd );
    }
    elsif ( $radio eq "Display Customer Data" ) {
        $cmd  = "/usr/bin/java -Djava.security.egd=file:/dev/./urandom BookStore2 'Display Customer Data'";
        print ( $cmd . "\n\n" );
        system( $cmd );
    }

    print <<EndofHTML;
        <form>
         <input type="button" value=" Back " onclick="history.go(-1);return false;" />
        </form>
       </b></font>
      </td>
     </tr>
    </table>
   </body>
  </html>
EndofHTML
}
elsif ( $act eq "HTML source" ) {
    print ( "Content-type: text/plain\n\n" );
    $cmd  = "/usr/bin/lynx -dump -source " . $ENV{HTTP_REFERER};
    $cmd .= "; echo \n\n\n\n";
    system( $cmd );
}
elsif ( $act eq "CGI source" ) {
  # Print plain text.
    print ( "Content-type: text/plain\n\n" );
    system( "/bin/cat BookStore.cgi; echo \n\n\n\n" );
}
elsif ( $act eq "Perl source" ) {
    print ( "Content-type: text/plain\n\n" );
    system( "/bin/cat BookStore.pl; echo \n\n\n\n" );
}
elsif ( $act eq "Java source" ) {
    print ( "Content-type: text/plain\n\n" );
    system( "/bin/cat BookStore.java; echo \n\n\n\n" );
}
else {
    print( "Content-type: text/html\n\n" );
    print( "No such option: <em>$act</em>" );
}
